Lib/multiprocessing/synchronize.py

```diff
@@ -103,7 +103,7 @@ def __getstate__(self):
         if sys.platform == 'win32':
             h = context.get_spawning_popen().duplicate_for_child(sl.handle)
         else:
-            if self.is_fork_ctx:
+            if getattr(self, "is_fork_ctx", False):
```

Comment:
Why not instead set `is_fork_ctx` in `__setstate__` below?

Reply:
I'm happy to change this. My question would be: is it guaranteed that either `__init__()` or `__setstate__()` will be called for every object? Or are there other ways the object might be created that would end up not calling either of these methods?

If there is no other way, I can update `__setstate__` but if there are other ways the objects could be created, this pattern will cover them all.

Let me know which one is best and I can update the PR accordingly!

Reply:
Perhaps grepping through uses of `SemLock` in the Python stdlib would give an insight? This is not a public API, so external use should not be relevant.

Reply:
Ok, so for my general questoin: for any python object, is it possible to create it without calling either `__init__()` or `__setstate__()`?
The answer is definitely yes as the user can do whatever they want with calling `__new__` directly.

In this particular, case, since SemLock is the base class for many public API (and thus user might create these objects arbitrarily) it is best to make `__getstate__` resilient to bad class creation.